### PR TITLE
Upgrade cops 1.0.1 => 1.1.3 + DSM 7

### PR DESCRIPTION
**Pull request by smaarn**
## Description

Two objectives:

1. Bring cops to DSM 7
2. Upgrade cops (while we're at it

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [x] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [x] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Package update
- [x] Includes small framework changes

Binaries are available here: https://github.com/smaarn/spksrc/releases/tag/cops-1.1.3-rc-2

### CODE REVIEW

**Comment by smaarn** [comment]
There is an open question regarding the beta flag: personally I would flag it as non beta on DSM 7 since the PHP configuration is really native.

But then that's just me :)

File: spk/cops/src/service-setup.sh
Comment (th0ma7) [inline spk/cops/src/service-setup.sh:74]: Is this still relevant?

**Review by th0ma7 — APPROVED** [review]
LGTM and indeed, the beta flag probably can be removed along with updating https://github.com/SynoCommunity/spksrc/issues/4524

File: spk/cops/src/service-setup.sh
Comment (smaarn) [reply spk/cops/src/service-setup.sh:74]: Good point. Given the comment I guess it can be removed safely.

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: So I removed everything which sounded like DSM 4 specifics.

The only thing is that I don't own a DSM 6 to validate it still works on those :/

Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: I got one, let me know the specifics you'd need testing on and I'll check it out.

Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: Here's a plain default install result (e.g. I've only hit next, no kobo selection):
Capture d’écran du 2022-05-01 07-33-40

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: Have you checked the path specified ? Does it match an existing directory ?

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: Regarding the PHP configuration I guess I should be documenting it and adding this as part of the DSM 6 wizard specifics. (It's a matter of enabling the right extensions)

Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: > Have you checked the path specified ? Does it match an existing directory ?

Well there is no `/volume1/calibre` directory... I presumed (perhaps wrongly) that it was going to generate one if non-existant...

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: Actually it's supposed to be the path to an existing calibre directory... This may sound awkward but, as of now, it doesn't create any folder structure.

Just updated the wizard to explicitely mention the list of extensions to enable.

Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: > Actually it's supposed to be the path to an existing calibre directory... 

I haven't noticed if this was clearly mentionned.  Couldn't it generate an "empty" db file in case it's non-existant?  On the other hand, I don't know exactly what this app is for nor how to use it so take this comment with a grain of salt.

Otherwise, if you want me to test that out could you provide me with an default db file that I could confirm testing with on DSM6?  Perhaps that "empty" db file could actually be provided with the package... just a thought.

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: > > Actually it's supposed to be the path to an existing calibre directory...
> 
> I haven't noticed if this was clearly mentionned. Couldn't it generate an "empty" db file in case it's non-existant? On the other hand, I don't know exactly what this app is for nor how to use it so take this comment with a grain of salt.

The idea of this app is to come as a companion to an existing calibre installation which would be handling the writing and organizing part. cops is "merely" a browser component which integrates smoothly with web browsers and is compliant with other ebook readers (for example Kobo ones, and it happens I use one :) ).

I believe we could generate an empty DB but I'm not sure of the added value here (nor that it would be maintainable). Will be checking.

> 
> Otherwise, if you want me to test that out could you provide me with an default db file that I could confirm testing with on DSM6? Perhaps that "empty" db file could actually be provided with the package... just a thought.

That's for sure. I'll attach one to the PR right away.

**Comment by smaarn** [comment]
Attaching an "empty" library folder: 
[calibre.zip](https://github.com/SynoCommunity/spksrc/files/8599111/calibre.zip)

File: spk/cops/src/service-setup.sh
Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: Did a git pull and rebuilt the package.  It's nice to mention which PHP extensions are needed but having a URL to point for help would be beneficial as otherwise the user will ask himself : am I missing something?  We could probably create a WebStation wiki page on our spksrc github and include our finding for creating packages and al (just a thought).

Additionally, perhaps you should include an additional `SPK_DEPENDS` for php?

An error occured while changing permissions:
```
2022/05/01 15:58:31	===> Step postinst. USER=http GROUP= SHARE_PATH=
2022/05/01 15:58:31	Begin save_wizard_variables
2022/05/01 15:58:31	End save_wizard_variables
2022/05/01 15:58:31	Begin service_postinst
2022/05/01 15:58:31	ACL version: 1 
2022/05/01 15:58:31	Archive: has_ACL,is_support_ACL 
2022/05/01 15:58:31	Owner: [th0ma7(user)] 
2022/05/01 15:58:31	--------------------- 
2022/05/01 15:58:31		 [0] group:administrators:allow:rwxpdDaARWc--:fd--  (level:0)
2022/05/01 15:58:31	Granting '' group permissions on /volume1/calibre
2022/05/01 15:58:31	(synoacltool.c, 125)No such Group 
2022/05/01 15:58:31	Granting '' group basic permissions on /volume1/calibre
2022/05/01 15:58:31	(synoacltool.c, 125)No such Group 
2022/05/01 15:58:32	End service_postinst
2022/05/01 15:58:32	install cops 1.1.3-6 End postinst ret=[0]
2022/05/01 15:58:32	install cops 1.1.3-6 Begin /bin/rm -rf /volume1/@tmp/pkginstall
2022/05/01 15:58:32	install cops 1.1.3-6 End /bin/rm -rf /volume1/@tmp/pkginstall ret=[0]
2022/05/01 15:58:36	install cops 1.1.3-6 Begin start-stop-status start
2022/05/01 15:58:36	install cops 1.1.3-6 End start-stop-status start ret=[0]
```
This leads to the same output as previously (e.g. `Database error`) as it can't find the db file, and I assume that's the reason why.

Comment (th0ma7) [reply spk/cops/src/service-setup.sh]: Re-tested but this time I made sure to create a `/volume1/calibre` folder using the GUI and that ACL was enabled.  Errors are now different which makes me think you would be better off adding a warning that ACL must be turned on.

Errors:
```
2022/05/03 10:05:24	===> Step postinst. USER=http GROUP= SHARE_PATH=
2022/05/03 10:05:24	Begin save_wizard_variables
2022/05/03 10:05:24	End save_wizard_variables
2022/05/03 10:05:24	Begin service_postinst
2022/05/03 10:05:24	Granting '' group permissions on /volume1/calibre
2022/05/03 10:05:24	(synoacltool.c, 125)No such Group 
2022/05/03 10:05:24	Granting '' group basic permissions on /volume1/calibre
2022/05/03 10:05:24	(synoacltool.c, 125)No such Group 
2022/05/03 10:05:24	End service_postins
```

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: Actualy this is related to another issue (which I will be fixing).

But agree on the permissions message to add !

Comment (smaarn) [reply spk/cops/src/service-setup.sh]: Added permissions related message + fixed the GROUP logic to not have the failures reported

**Comment by smaarn** [comment]
Rebased @ head.

**Comment by th0ma7** [comment]
@smaarn cycles have been limited lately.  Is there a need for another review before merging or where you good to go?

**Comment by smaarn** [comment]
> @smaarn cycles have been limited lately. Is there a need for another review before merging or where you good to go?



> @smaarn cycles have been limited lately. Is there a need for another review before merging or where you good to go?

@th0ma7 thanks for circling back, and no worries I had no cycles available lately so can't complain :D 

Actually an extra review could be appreciated regarding the pseudo fragment mechanism used for the wizards (basically they are brutally concatenated). Since it can be seen as pretty ugly just wanted to know if it was ok.

**Review by th0ma7 — APPROVED** [review]
LGTM

**Comment by th0ma7** [comment]
@smaarn Let me give it a shot on some of my NAS over the week-end to confirm it works all-right.  Now I need to find back the metadata :)

BTW, I see you have a really good handle on WebStation type packages.  Perhaps you could help me out to make rutorrent compatible on DSM7 part of your free-time?  I could then update it to the latest version.

**Comment by smaarn** [comment]
> @smaarn as you use a _resource worker_ to define a package specific php profile with the required extensions enabled, what is the wizard page for, that tells to activate php extensions?
> 
> BTW: resource worker works with DSM 6 and DSM 7.

I'm not sure I'm understanding what you mean. The resource worker doesn't require in itself any wizard page, it's basically replacing what you can configure through Web Station and PHP profiles (which is documented in the installation wizard of DSM 6 on how to do it manually), was it your question ?

Unfortunately the web resource worker don't work in DSM 6 from what I recall.

**Comment by smaarn** [comment]
> @smaarn Let me give it a shot on some of my NAS over the week-end to confirm it works all-right. Now I need to find back the metadata :)
> 
> BTW, I see you have a really good handle on WebStation type packages. Perhaps you could help me out to make rutorrent compatible on DSM7 part of your free-time? I could then update it to the latest version.

Thanks for your help !

I'll try having a look at rutorrent but it may prove to be a little bit more complex since it requires a set of authorization mechanisms which I'm not totally operational with (basically some folders need to be both writable for the web part and the non web part... I remember last time I played with it it was kind of a nightmare :) ).

Things are going a little bit more quiet over here so I may have some cycles over the coming weeks to check it.

**Comment by smaarn** [comment]
Squashing all the commits after rebase at head in order to be able to merge by EOD (CET time) if everything goes fine.

**Comment by mreid-tt** [comment]
@hgy59, I was trying to publish some packages that were previously merged. I noted that for the `noarch-6.0` version I was getting an upload error (409) which suggests that it was already there but not visible in the admin interface. After some checks it seems that in the INFO file there is a line `firmware="3.1-1594"` which is not appropriate for this build. Need some assistance in fixing this in the framework.

**Comment by hgy59** [comment]
expected versions in INFO file
- firmware="3.1-1594"
- os_min_ver="6.1-15047"

This is correct with my scripts to build and publish noarch packages

make_all_noarch.sh
```bash
#!/bin/bash

# make all noarch packages
echo ""
echo "===> MAKE noarch (default = DSM 3.1) <==="
make
echo ""
echo "===> MAKE noarch for SRM 1.1  <==="
make TCVERSION=1.1
echo ""
echo "===> MAKE noarch for DSM 6.1  <==="
make TCVERSION=6.1
echo ""
echo "===> MAKE noarch for DSM 7.0  <==="
make TCVERSION=7.0
echo ""
```

publish_all_noarch.sh
```bash
#!/bin/bash

# publish all noarch packages
echo ""
echo "===> PUBLISH noarch (default = DSM 3.1) <==="
make publish
echo ""
echo "===> PUBLISH noarch for SRM 1.1  <==="
make publish TCVERSION=1.1
echo ""
echo "===> PUBLISH noarch for DSM 6.1  <==="
make publish TCVERSION=6.1
echo ""
echo "===> PUBLISH noarch for DSM 7.0  <==="
make publish TCVERSION=7.0
echo ""
```

**Comment by hgy59** [comment]
@mreid-tt I have deleted the two packages of cops 1.1.3 now, so you can restart the deployment.

**Comment by mreid-tt** [comment]
> expected versions in INFO file
> 
> * firmware="3.1-1594"
> * os_min_ver="6.1-15047"

hmm, recall I don't have a local build instance and only publish via GitHub. Would a possible solution be to update the matrix in this line:

https://github.com/SynoCommunity/spksrc/blob/c817629413227c4ce4073a2908dd23e6e2585206/.github/workflows/build.yml#L84

Perhaps changing `noarch-6.0` to `noarch-6.1`? Not sure if this would resolve the GitHub based builds...

**Comment by hgy59** [comment]
@mreid-tt indeed it was the 6.0 vs 6.1 version
see lines 61-63 in mk/spksrc.spk.mk
```
else ifeq ($(call version_ge, $(TCVERSION), 6.1),1)
SPK_TCVERS = dsm6
TC_OS_MIN_VER = 6.1-15047
```

building  with TCVERSION=6.0 creates the same package as without TCVERSION (i.e. the same as `noarch`)

it is now fixed on head on master branch.

**Comment by mreid-tt** [comment]
> @mreid-tt indeed it was the 6.0 vs 6.1 version [---snip---]
> building with TCVERSION=6.0 creates the same package as without TCVERSION (i.e. the same as `noarch`)
> 
> it is now fixed on head on master branch.

Thanks so much for this @hgy59, thanks for merging the fix into the master branch.

> @mreid-tt I have deleted the two packages of cops 1.1.3 now, so you can restart the deployment.

I was able to run a new job and all builds completed successfully. The package has now been published.

**Pull request merged**
